cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -50,13 +50,14 @@ class BaseTestStringKernels : public ::testing::Test {
     CheckScalarUnary(func_name, type(), json_input, out_ty, json_expected, options);
   }
 
-  void CheckBinaryScalar(std::string func_name, std::string json_left_input,
-                         std::string json_right_scalar, std::shared_ptr<DataType> out_ty,
-                         std::string json_expected,
-                         const FunctionOptions* options = nullptr) {
-    CheckScalarBinaryScalar(func_name, type(), json_left_input, json_right_scalar, out_ty,
-                            json_expected, options);
-  }
+  // void CheckBinaryScalar(std::string func_name, std::string json_left_input,
+  //                        std::string json_right_scalar, std::shared_ptr<DataType>
+  //                        out_ty, std::string json_expected, const FunctionOptions*
+  //                        options = nullptr) {
+  //   CheckScalarBinaryScalar(func_name, type(), json_left_input, json_right_scalar,
+  //   out_ty,
+  //                           json_expected, options);
+  // }
```

Comment:
Why is this commented out?

Reply:
This was existing code that was not used. I need to provide better support for binary kernels, so that scalar-array/array-scalar can be tested.